locking.js

```diff
@@ -60,7 +54,6 @@ async function acquire(config, state, task) {
     if (config.no_locking) return true;
 
     assert(task);
-    assert(task.resources);
```

Comment:
Why can this assertion be removed, but the following can't?

Reply:
With this PR `task.resources` will always be an array. If it is empty we can bail out early. Otherwise we'd run the `external_locking.externalAcquire()` branch a few lines below.